### ISSUE TITLE
v3 not uploading any files

### ISSUE THREAD

**Issue by aaronmeder** [SamKirkland/FTP-Deploy-Action#42]
**Bug Description**
I cannot get it to upload any files when using v3. 
v2 works great.

I have a nuxt app that I first build / generate and then want to upload the dist files.
The nuxt/dist/ directory (which contains the generated files that should be uploaded) is empty in my repo and gets filled during the generate step.

**My Action Config**
```yaml
name: Deploy to Server

on: [push]

jobs:
  build:

    runs-on: ubuntu-latest

    steps:
      
    - uses: actions/checkout@master
      with:
        fetch-depth: 2
    - name: Use Node.js 10.x
      uses: actions/setup-node@v1
      with:
        node-version: 10.x

    - name: Install npm dependencies
      run: npm ci
      working-directory: ./nuxt

    - name: Generate nuxt app
      run: npm run generate && ls -l dist/
      working-directory: ./nuxt

    - name: Deploy to FTP
      uses: SamKirkland/FTP-Deploy-Action@3.0.0
      with:
        FTP_SERVER: ${{ secrets.FTP_SERVER }}
        FTP_USERNAME: ${{ secrets.FTP_USERNAME }}
        FTP_PASSWORD: ${{ secrets.FTP_PASSWORD }}
        LOCAL_DIR: nuxt/dist
```

**My Action Log**

The log starts with the directory listing at the end of stepp "Generate nuxt app" - that would be the files that need to be uploaded.
```
2020-02-19T07:36:25.6855059Z -rw-r--r-- 1 runner docker  3584 Feb 19 07:36 200.html
2020-02-19T07:36:25.6855828Z -rw-r--r-- 1 runner docker   435 Feb 19 07:36 README.md
2020-02-19T07:36:25.6856745Z drwxr-xr-x 3 runner docker  4096 Feb 19 07:36 _nuxt
2020-02-19T07:36:25.6857368Z -rw-r--r-- 1 runner docker  1393 Feb 19 07:36 favicon.ico
2020-02-19T07:36:25.6857884Z -rw-r--r-- 1 runner docker 12699 Feb 19 07:36 icon.png
2020-02-19T07:36:25.6858414Z -rw-r--r-- 1 runner docker  5138 Feb 19 07:36 index.html
2020-02-19T07:36:25.6858930Z -rw-r--r-- 1 runner docker   995 Feb 19 07:36 sw.js
2020-02-19T07:36:25.7065144Z ##[group]Run SamKirkland/FTP-Deploy-Action@3.0.0
2020-02-19T07:36:25.7065281Z with:
2020-02-19T07:36:25.7065717Z   ftp-server: ***
2020-02-19T07:36:25.7065968Z   ftp-username: ***
2020-02-19T07:36:25.7066120Z   ftp-password: ***
2020-02-19T07:36:25.7066217Z   local-dir: nuxt/dist
2020-02-19T07:36:25.7066315Z ##[endgroup]
2020-02-19T07:36:25.7091823Z ##[command]/usr/bin/docker run --name e87b5214e197375c4c41d4848da5bf6c35ced1_5bdb6c --label e87b52 --workdir /github/workspace --rm -e INPUT_FTP-SERVER -e INPUT_FTP-USERNAME -e INPUT_FTP-PASSWORD -e INPUT_LOCAL-DIR -e INPUT_GIT-FTP-ARGS -e HOME -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e GITHUB_ACTIONS=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/nuxt-actions-starter/nuxt-actions-starter":"/github/workspace" e87b52:14e197375c4c41d4848da5bf6c35ced1
2020-02-19T07:36:32.6177059Z ##[group]Uploading files
2020-02-19T07:36:32.6205728Z [command]git ftp push --force --auto-init --verbose --syncroot nuxt/dist --user *** --passwd *** ***
2020-02-19T07:36:32.6380416Z Wed Feb 19 07:36:32 UTC 2020: git-ftp version 1.5.1 running on Linux fe536c023aab 5.0.0-1031-azure #33-Ubuntu SMP Thu Feb 6 22:26:13 UTC 2020 x86_64 GNU/Linux
2020-02-19T07:36:32.6380665Z Wed Feb 19 07:36:32 UTC 2020: Forced mode enabled.
2020-02-19T07:36:32.6380865Z Wed Feb 19 07:36:32 UTC 2020: Auto init if needed.
2020-02-19T07:36:32.6404687Z Wed Feb 19 07:36:32 UTC 2020: Using syncroot nuxt/dist if exists.
2020-02-19T07:36:32.6859517Z Wed Feb 19 07:36:32 UTC 2020: Host is '***'.
2020-02-19T07:36:32.6869996Z Wed Feb 19 07:36:32 UTC 2020: User is '***'.
2020-02-19T07:36:32.6902666Z Wed Feb 19 07:36:32 UTC 2020: Password is set.
2020-02-19T07:36:32.6955616Z Wed Feb 19 07:36:32 UTC 2020: CACert is ''.
2020-02-19T07:36:32.6990568Z Wed Feb 19 07:36:32 UTC 2020: Insecure is '0'.
2020-02-19T07:36:32.7037587Z Wed Feb 19 07:36:32 UTC 2020: Proxy is ''.
2020-02-19T07:36:32.7095137Z Wed Feb 19 07:36:32 UTC 2020: Protocol not set, using default protocol ftp://.
2020-02-19T07:36:32.7144358Z Wed Feb 19 07:36:32 UTC 2020: Path is ''.
2020-02-19T07:36:32.7174846Z Wed Feb 19 07:36:32 UTC 2020: Syncroot is 'nuxt/dist/'.
2020-02-19T07:36:32.7207604Z Wed Feb 19 07:36:32 UTC 2020: The remote sha1 is saved in file '.git-ftp.log'.
2020-02-19T07:36:32.7219907Z Wed Feb 19 07:36:32 UTC 2020: Check if curl is functional.
2020-02-19T07:36:32.7312207Z Wed Feb 19 07:36:32 UTC 2020: Retrieving last commit from ftp://***:***@***/.
2020-02-19T07:36:32.9381498Z #=#=#                                                                         
2020-02-19T07:36:33.1271264Z ##O#- #                                                                       
2020-02-19T07:36:33.3446146Z ##O=#  #                                                                      
2020-02-19T07:36:33.5317988Z #=#=-#  #                                                                     
2020-02-19T07:36:33.7187990Z -#O#- #   #                                                                   
2020-02-19T07:36:33.9063575Z -=#=#   #   #                                                                 
2020-02-19T07:36:33.9063751Z 
2020-02-19T07:36:33.9064058Z ######################################################################## 100.0%
2020-02-19T07:36:34.0081190Z Wed Feb 19 07:36:34 UTC 2020: Unknown SHA1 object, could not determine changed files, taking all files.
2020-02-19T07:36:34.0114752Z Wed Feb 19 07:36:34 UTC 2020: No files to sync. All changed files ignored.
2020-02-19T07:36:34.0128108Z Wed Feb 19 07:36:34 UTC 2020: There are no files to sync.
2020-02-19T07:36:34.0143167Z Wed Feb 19 07:36:34 UTC 2020: Uploading commit log to ftp://***:***@***/.git-ftp.log.
2020-02-19T07:36:34.2285280Z #=#=#                                                                         
2020-02-19T07:36:34.4164336Z ##O#- #                                                                       
2020-02-19T07:36:34.6340493Z ##O=#  #                                                                      
2020-02-19T07:36:34.8210987Z #=#=-#  #                                                                     
2020-02-19T07:36:35.0078690Z -#O#- #   #                                                                   
2020-02-19T07:36:35.1959372Z -=#=#   #   #                                                                 
2020-02-19T07:36:35.1960195Z -=O#- #  #    #                                                               
2020-02-19T07:36:35.2915372Z Wed Feb 19 07:36:35 UTC 2020: Last deployment changed from d47023f9a64196661463328f32be24d54f651ef0 to 4acf835b1700e378dbfe94fd5589356c4aa54514.
2020-02-19T07:36:35.2963181Z ##[endgroup]
```

tl;dr: error always says "No files to sync. All changed files ignored." 🤷🏻‍♂️

Don't know if it's a bug or I am doing something wrong.

Thanks for help.
Aaron

**Comment by SamKirkland** [comment]
Hey Aaron,

Version 3 will only upload tracked files by default (files committed to your github project). The reason for this change was so only diffs are uploaded rather than all files on every publish. This reduces publish time for large websites from hours to minutes but requires more setup initially.

To resolve this you must create a `.git-ftp-include` file in the root of your project with the content `!nuxt/dist/` so the folder is always uploaded. Or you can commit that file to git and only diffs will be published. [See docs for more information](/SamKirkland/FTP-Deploy-Action#build-and-publish-reactangularvue-website)

I'm going to close out this issue, let me know if this doesn't resolve the issue and i'll reopen.

**Comment by aaronmeder** [comment]
Thanks so much @SamKirkland !
Somehow I missed that in the docs.
Thank you for the great tool.
